onnxruntime/test/common/tensor_op_test_utils.cc

```diff
@@ -14,5 +14,10 @@ RandomValueGenerator::RandomValueGenerator(optional<RandomSeedType> seed)
       output_trace_{__FILE__, __LINE__, "ORT test random seed: " + std::to_string(random_seed_)} {
 }
 
+FixedPatternValueGenerator::FixedPatternValueGenerator()
+    : generator_{0},
+      output_trace_{__FILE__, __LINE__, "ORT test random seed with fixed pattern tensor generator"} {
```

Comment:
nit: I think this scoped trace is not really necessary. in the case where there is a random seed with a dynamic value, it is more useful since it shows that value when there is a test failure while it is in scope.

Reply:
I feel it's needed. If we tell what seed is used, we should also tell when seed is not used. I hope to propagate the error structure as clearer as possible.